### PR TITLE
[Net] Rename RPC "puppet" to "auth", drop "master".

### PR DESCRIPTION
**Pull request by Faless**
In this PR:

The first commit completely removes the `RPC_MODE_MASTER` ("master" keyword), and renames the `RPC_MODE_PUPPET` to **`RPC_MODE_AUTHORITY`** (`@rpc(auth)` annotation, default).

The first commit also renames the "Node.[get|set]_network_master" methods to "Node.[get|set]_network_authority".

The first commit also renames the `RPC_MODE_REMOTE` constant to `RPC_MODE_ANY`.

`RPC_MODE_MASTER` in Godot 3.x meant that a given RPC would be callable by any puppet peer on the master, while `RPC_MODE_PUPPET` meant that it would be callable by the master on any puppet.

Beside proving to be very confusing to the user (referring to where it could be called instead of who can call it) the `RPC_MODE_MASTER` is quite useless. It is almost the same as `RPC_MODE_REMOTE` (anyone can call) with the exception that the network master cannot.

While this could be useful to check in some case, in such a function you would anyway need to check
in code who is the caller via `get_rpc_sender_id()`, so adding the check there for those rare cases does not warrants a dedicated mode.

---

This PR also slightly improve the GDScript `@rpc` annotation to allow any order for the string arguments.

### CODE REVIEW

File: core/io/multiplayer_api.h [hunk 1]
```diff
@@ -41,9 +41,8 @@ class MultiplayerAPI : public RefCounted {
 public:
 	enum RPCMode {
 		RPC_MODE_DISABLED, // No rpc for this method, calls to this will be blocked (default)
-		RPC_MODE_REMOTE, // Using rpc() on it will call method in all remote peers
-		RPC_MODE_MASTER, // Using rpc() on it will call method on wherever the master is, be it local or remote
-		RPC_MODE_PUPPET, // Using rpc() on it will call method for all puppets
+		RPC_MODE_REMOTE, // Any peer can call this rpc()
```
Comment (akien-mga) [inline core/io/multiplayer_api.h:44]: Should this be renamed `RPC_MODE_ANY` to match the annotation?
It also seems tautological for a "Remote Procedure Call Mode" to be "Remote" :)

Comment (akien-mga) [reply core/io/multiplayer_api.h:44]: That being said it might be ambiguous and understood as "any mode", i.e. "both Disabled and Master".

Comment (mhilbrunner) [reply core/io/multiplayer_api.h:44]: No, RPC_MODE_ANY would be fine I think.

**Comment by fire** [comment]
Other than the merge conflicts, what is impeding approval of this pr?

**Review by zerastos — APPROVED** [review]

File: core/io/multiplayer_api.h (hunk 1 above)
Comment (Faless) [reply core/io/multiplayer_api.h]: done

**Comment by Faless** [comment]
Updated the RPC to `auth` (`RPC_MODE_AUTHORITY`), as discussed in `#networking` meeting of *29th Aug 2021*.

**Comment by Faless** [comment]
This PR needs one github approval to be merged, but was approved in the `#networking` meeting of `29th Aug 2021`.

**Review by fire — APPROVED** [review]
Was in the meeting for networking. Approving because the networking team approved as a whole.

**Pull request merged**